homework-6/script-1.js

```diff
@@ -0,0 +1,78 @@
+//Написати функцію, яка приймає 4 аргументи і шукає найменше число серед них
+function getMinNumber(num_1, num_2, num_3, num_4) {
+    const result = Math.min(num_1, num_2, num_3, num_4);
+    if (!isNaN(result)) {
+        return result;
+    } else {
+        console.log("I can't figure out min number");
+    }
+}
+getMinNumber("5", 0, 5, 2);
+
+/*Написати функцію, яка приймає 2 аргументи та шукає найбільший спільний дільник для них. 
+Якщо такого числа немає, то повідомляти про відсутність*/
+function getGreatestCommonDivisor(num_1, num_2) {
+    if (num_2 > num_1) {
+        return getGreatestCommonDivisor(num_2, num_1);
+    } else if (!num_2) {
+        return num_1;
+    } else if (num_1 > num_2) {
+        return getGreatestCommonDivisor(num_2, num_1 % num_2);
+    } else {
+        console.log("The greatest common divisorr is missing");
+    }
+}
+getGreatestCommonDivisor(25, 5);
+
+//Написати функцію, яка приймає 1 аргумент та перевіряє чи дане число ідеальне
+function getPerfectNumber(num) {
```

Comment:
There is no validation at all, I can pass any value and the function will say that is not a perfect number, even if it is `undefined` or `NaN`
<img width="566" alt="image" src="https://user-images.githubusercontent.com/84010501/191335794-2e4b8673-7313-4f7d-b101-4492400fab5f.png">